### PR TITLE
[Code Quality - Cover]: Remove unnecessary `temporaryMinHeight`

### PR DESCRIPTION
**Pull request by ntsekouras**
While investigating a bug in `Cover` block I saw that besides the 800 lines of the `edit` function that makes it really hard to follow the logic, there are also some weird(to me at least) things that make it even harder.

I think this block needs a bit cleanup and I'm starting with this PR which is intentionally small, to be better tested for regressions.

In this PR I'm removing the usage of `withDispatch` and the unused `withInstanceId`, but most importantly I'm removing an extra variable used for minHeight in various places (`temporaryMinHeight`). I'm guessing this was for not updating the main attribute on resize, in order to not create `undo` levels in history, but that's not the case here, since we don't mark consecutive quick changes of the same block attribute for history in the core data store.

## Testing instructions
Everything in Cover block should work exactly as before for either existing or new Cover blocks.

### CODE REVIEW

File: packages/block-library/src/cover/edit.js [hunk 1]
```diff
@@ -309,8 +309,6 @@ function CoverEdit( {
 	overlayColor,
 	setAttributes,
 	setOverlayColor,
-	toggleSelection,
```
Comment (Mamaduka) [inline packages/block-library/src/cover/edit.js:312]: The `toggleSelection` is available as a block edit component prop; I'm not sure why we were overriding it.

https://github.com/WordPress/gutenberg/blob/11b0d32adc300dae20d00228927a2719cafb53f2/packages/block-editor/src/components/block-list/block.js#L115-L117

**Comment by Mamaduka** [comment]
Thanks, Nik.

I've not tested this yet, but this is a great start. The cover block needs some refactoring.

File: packages/block-library/src/cover/edit.js (hunk 1 above)
Comment (ntsekouras) [reply packages/block-library/src/cover/edit.js:312]: Good catch! I guess it might not be available at the time this was written? 🤔 Also it's easy to miss some of these properties that are available 😄 I updated.

**Comment by ramonjd** [comment]
This is looking GTM! Thank you!

https://user-images.githubusercontent.com/6458278/154621565-5a52a86e-e875-42b3-a346-921f2ddf32f4.mp4

> The cover block needs some refactoring.

`core/cover-2` anyone? 😆

**Review by ramonjd — APPROVED** [review]
👍
